### ISSUE TITLE
py-pytorch-regression-boston-housing

### ISSUE THREAD

**Issue by germanmandrini** [daines-analytics/deep-learning-projects#1]
Hi, I'm learning to use pytorch and using this script as an example. 
It seems to me that it provides always the same prediction. I change the values of the "row" and the prediction is the same. I also modified the evaluate_model function to print yhat, and they are always the same. Could it be something wrong with the code or I'm not understanding how it works?
Thanks

The script is:
https://github.com/daines-analytics/deep-learning-projects/blob/master/py-pytorch-regression-boston-housing/py-pytorch-regression-boston-housing.ipynb

**Comment by daines-analytics** [comment]
Not seeing what changes you have made to the scripts and what outcomes you were hoping to see, I can only make one general comment at this time. Have you tried to modify the seed number in code block #4 to see whether the script might have generated different predictions and results?

**Comment by germanmandrini** [comment]
After training a model, If I change the values in row, I get always the same yhat:

row = [0.00632,18.00,2.310,0,0.5380,6.5750,65.20,4.0900,1,296.0,15.30,396.90,4.98]
yhat = predict(row, model)

Seems like the model learns one average value and always predict that one. Is that possible? Do you get different results for different predictors?

**Comment by daines-analytics** [comment]
I see what you were trying to do. I will re-create the environment and run the script again to check it out. It is possible there is a typo or mistake somewhere in the script.

**Comment by daines-analytics** [comment]
I tried a few things, and I think I found what was happening. The model appeared to have been stuck on some local minimum and could not get out of it. I switched the layer activation function from SIgmoid to ReLU as well as the optimizer from SGD to Adam. The new model now has a better overall RMSE and different predictions for different input rows. I hope this helps.

**Comment by germanmandrini** [comment]
That fixed it. Than you!
